src/storage/sequelize.ts

```diff
@@ -142,8 +147,8 @@ export class SequelizeStorage implements UmzugStorage {
 				tableName: this.tableName,
 				schema: this.schema,
 				timestamps: this.timestamps,
-				charset: 'utf8',
-				collate: 'utf8_unicode_ci',
+				charset: DIALECTS_WITH_CHARSET_AND_COLLATE.has(this.sequelize.dialect!.name!) ? 'utf8' : null,
+				collate: DIALECTS_WITH_CHARSET_AND_COLLATE.has(this.sequelize.dialect!.name!) ? 'utf8_unicode_ci' : null,
```

Comment:
I agree with the the linter here. If dialect is an optional property we shouldn't use !.

Let's make it a variable and null check it:

```ts
const dialectName = this.sequelize.dialect?.name
const hasCharsetAndCollate = dialectName && DIALECTS_WITH_CHARSET_AND_COLLATE.has(dialectName)

// ...

charset: hasCharsetAndCollate ? 'utf8' : null,
collate: hasCharsetAndCollate ? 'utf8_unicode_ci' : null,
```